client/src/components/ElementEditor/Toolbar.js

```diff
@@ -1,15 +1,28 @@
-import React, { PureComponent } from 'react';
+import React, { PureComponent, PropTypes } from 'react';
+import { inject } from 'lib/Injector';
 
 // eslint-disable-next-line react/prefer-stateless-function
 class Toolbar extends PureComponent {
   render() {
+    const { AddNewButtonComponent, elementTypes } = this.props;
     return (
-      <div />
+      <div>
+        <AddNewButtonComponent elementTypes={elementTypes} />
+      </div>
     );
   }
 }
 
 Toolbar.defaultProps = {};
-Toolbar.propTypes = {};
+Toolbar.propTypes = {
+  elementTypes: PropTypes.array.isRequired,
+  AddNewButtonComponent: PropTypes.oneOfType([PropTypes.node, PropTypes.func]).isRequired,
+};
 
-export default Toolbar;
+export default inject(
+  ['ElementAddNewButton'],
+  (AddNewButtonComponent) => ({
+    AddNewButtonComponent,
+  }),
+  () => 'ElementToolbar'
```

Comment:
Can you make this `ElementEditor.ElementToolbar` so it's under the context of being in an ElementEditor?

Reply:
Isn't this supposed to match the name which it's registered under Injector?

Reply:
Nah, this is the "context key" - essentially what this does is say "give me the add new button component, in the context of `ElementToolbar`." This allows devs to configure specific HOCs for `AddNewButtonComponent` in different contexts.

In this case it's important to highlight that the context is within `ElementEditor`